Topics/Tech_Stacks/Unity_ML_Agents.md

```diff
@@ -0,0 +1,199 @@
+# Introduction to Unity's ML-Agents Toolkit
+
+Unity's ML-Agents Toolkit, or simply ML Agents, is a package that provides a simple and intuitive way to develop agents capable of interacting with both 2D and 3D environments in Unity. It is a project that aims to bring machine learning together with the many environments that can be easily developed in Unity, and provide an easy and accessible way for game developers to work with machine learning.
+
+In this introductory tutorial, we will explore the basics of reinforcement learning with an example built completely from scratch.
+
+# Installation Guide
+Before installing Unity's ML-Agents package, ensure that you installed the following:
+
+[Unity 2022.3 or Later](https://unity3d.com/get-unity/download)
+
+[Python 3.10.12 or Higher](https://www.python.org/downloads/)
+
+PyTorch, it is recommended that you create a virtual environment to avoid conflicts with any existing packages.
+
+```pip3 install torch~=1.13.1 -f https://download.pytorch.org/whl/torch_stable.html```
+
+ML Agents Python Package, once again, install in a virtual environment.
+
+```pip3 install ./ml-agents-envs```
+
+```pip3 install ./ml-agents```
+
+Finally, within Unity's Package Manager, find and install the ```com.unity.ml-agents``` package.
+
+# What is Reinforcement Learning?
+
+In this tutorial we will use Unity's ML-Agents Toolkit to explore the basics of reinforcement learning. Reinforcement Learning is a form of machine learning where an agent will learn to make decisions by interacting with an environment. Based on the actions it performs, it will receive a reward that indicates how well it performed. Over time, the goal of the agent is to learn a strategy that maximizes the reward it receives.
```

Comment:
Cool! But what can you use this for?

Reply:
I expanded on the introduction to also talk about the possible uses of machine learning, especially in a unity environment. I feel adding it here kinda breaks the flow